core/node/libp2p/rcmgr.go

```diff
@@ -0,0 +1,375 @@
+package libp2p
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	config "github.com/ipfs/go-ipfs/config"
+	"github.com/ipfs/go-ipfs/repo"
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	rcmgr "github.com/libp2p/go-libp2p-resource-manager"
+	"go.uber.org/fx"
+)
+
+const NetLimitDefaultFilename = "limit.json"
+const NetLimitTraceFilename = "rcmgr.json.gz"
+
+var NoResourceMgrError = fmt.Errorf("missing ResourceMgr: make sure the daemon is running with Swarm.ResourceMgr.Enabled")
+
+func ResourceManager(cfg config.ResourceMgr) func(fx.Lifecycle, repo.Repo) (network.ResourceManager, Libp2pOpts, error) {
+	return func(lc fx.Lifecycle, repo repo.Repo) (network.ResourceManager, Libp2pOpts, error) {
+		var limiter *rcmgr.BasicLimiter
+		var manager network.ResourceManager
+		var opts Libp2pOpts
+
+		// Config Swarm.ResourceMgr.Enabled decides if we run a real manager
+		enabled := cfg.Enabled.WithDefault(false)
+
+		/// ENV overrides Config (if present)
+		switch os.Getenv("LIBP2P_RCMGR") {
+		case "0", "false":
+			enabled = false
+		case "1", "true":
+			enabled = true
+		}
+
+		if enabled {
+			log.Debug("libp2p resource manager is enabled")
+
+			repoPath, err := config.PathRoot()
+			if err != nil {
+				return nil, opts, fmt.Errorf("error opening IPFS_PATH: %w", err)
+			}
+
+			// Try defaults from limit.json if provided
+			// (a convention to make libp2p team life easier)
+			limitFilePath := filepath.Join(repoPath, NetLimitDefaultFilename)
+			_, err = os.Stat(limitFilePath)
+			if !errors.Is(err, os.ErrNotExist) {
+				limitFile, err := os.Open(limitFilePath)
+				if err != nil {
+					return nil, opts, fmt.Errorf("error opening limit JSON file %q: %w", limitFilePath, err)
+				}
+				defer limitFile.Close() //nolint:errcheck
+				limiter, err = rcmgr.NewDefaultLimiterFromJSON(limitFile)
+				if err != nil {
+					return nil, opts, fmt.Errorf("error parsing limit file: %w", err)
+				}
+
+			} else {
+				// Use defaults from go-libp2p
+				log.Debug("limit file %s not found, creating a default resource manager", NetLimitDefaultFilename)
+				limiter = rcmgr.NewDefaultLimiter()
```

Comment:
- [ ] hardcode these implicit defaults in go-ipfs sources + document them in `docs/config.md`
- [ ] (TBD) remove `limit.json` (one can copy&paste values from `docs/config.md` and update `$IPFS_PATH/config`)
  - @vyzo @marten-seemann how painful will be removal of `limit.json` logic above for you? we'd like to reduce the number of places where user can override the defaults.

Reply:
I ended up keeping `limit.json` and hiding `Swarm.ResourceMgr.Limits` for now.
Having a single source of truth makes RC testing easier.